fs/io_zenfs.cc

```diff
@@ -162,10 +172,13 @@ ZoneFile::ZoneFile(ZonedBlockDevice* zbd, std::string filename,
       fileSize(0),
       filename_(filename),
       file_id_(file_id),
-      nr_synced_extents_(0) {}
+      nr_synced_extents_(0),
+      c_time_(time(0)) {}
```

Comment:
This is changing the storage format of the file system! You need to make the change backward compatible (or step file system version, which does not seem motivated..)

Reply:
Ok. So if we don't want to bump the filesystem version then I can see 2 options.
     1. Do not error out if c_time_ is not found for a file, report 0(or epoch) for such files.
     2. Hold off this patch until we bump up the filesystem version, then, add a check to not error out if c_time_ is not found and if filesystem's "version_" is older than CURRENT_VERSION.

However, I found this timestamps to be useful for lot of purposes.
Let me know what you prefer.

Reply:
I think we can report 0/epoch. And make sure you implement Modification time, not creation time. The RocksDB file system interface only wants modification time, not creation time. Store the creation time and update the value on every sync. But do pull this patch out of this pull request and submit it separately.

Reply:
Ok.
Thanks Hans.